docs/developer-guide/testing.md

```diff
@@ -24,8 +24,44 @@ gobopt='--set ana.base.privatization write+lock' ./scripts/update_suite.rb
 ```
 
 ### Writing
-* Add parameters to a regression test in the first line: `// PARAM: --set warn.debug true`
-* Annotate lines inside the regression test with comments: `arr[9] = 10; // WARN`
+Regression tests use single-line comments (with `//`) as annotations.
+
+#### First line
+A comment on the first line can contain the following:
+
+| Annotation | Comment |
+| ---------- | ------- |
+| `PARAM: ` <br> (NB! space) | The following command line parameters are added to Goblint for this test. |
+| `SKIP` | The test is skipped (except when run with `./scripts/update_suite.rb group`). |
+| `NOMARSHAL` | Marshaling and unmarshaling of results is not tested on this program. |
+| `TERM` | The expected Goblint result is that the program terminates. |
+| `NONTERM` <br> or `NOTERM` | The expected Goblint result is that the program does not terminate. |
```

Comment:
I think this checked in terms of the end of main being reachable, which has very little to do with termination. I think we should rename this while we're documenting it.

Reply:
Well, it has something to do with very primitive termination analysis, although that check is not even sound because it doesn't account for early exit via `exit` or such. It shouldn't be too hard to make such a simple yet sound termination analysis.

Actually, I'll just remove these annotations and replace them with explicit checks for (un)reachability at the end of `main` like we check for (un)reachability elsewhere. Then there's no need to even rename it.

Reply:
Yes, our students are working on a simple yet sound termination analysis, though primarily with the goal to prover termination, whereas non-reachability of all exit points would be prove definite non-termination. Though I think that there are only few cases where that would work.